### DISCUSSION TITLE
MFE 2 discussions

### DISCUSSION THREAD

**Discussion by dongle-the-gadget** [MicaForEveryone/MicaForEveryone#356] [Ideas]
Wanted to make this discussion to be sort of a megathread for discussing ideas pertaining to Mica For Everyone 2, our (planned) release for (hopefully) the next version of Mica For Everyone.

One thing that we will get right out of the bag: MFE 2 will be using .NET 7 + WinUI 3, so unfortunately no fancy DWM hooking to create custom effects or anything. (Regardless, DWM hooking has a high chance of triggering anti-cheat, it seems).

With that out of the way, we also want to ask some questions:
- Would you be okay if we made breaking changes to the configuration file? For instance, switching to another language (like JSON) from XCL (which is our current one).
- Would you mind we move away from portable + unpackaged installers (EXE and MSI installers) to packaged installers (MSIX)? If not, would you install a certificate to install MFE, or is a valid codesigning certificate a requirement for you?
- How do you feel about a headless (basically no user-interacting parts, you just manually write out the rules in a text editor) version of Mica For Everyone?

**Comment by tilsgee** [comment]
Well, you do you. Except MSIX part. Can you please, think about LTSC users?. LTSC users didn't have capability to install MSIX easily, let alone APPX file

**Reply by tilsgee** [reply]
Also, it's faster to update from MSI/EXE installer, instead of waiting for Microsoft approval just to add 10 lines of code for an update 

TranslucentTB suffered from this issue, btw

**Reply by Void48** [reply]
Sorry for the late reply, but I'm pretty sure it's possible to install WinGet (App Installer app) on recent versions of LTSC, so this shouldn't really be an issue in my opinion. The majority of people are also not using LTSC and if you are, you should just update to the latest LTSC version tbh

**Comment by Undisputed00x** [comment]
https://github.com/MicaForEveryone/MicaForEveryone/discussions/366#discussion-5682463

**Comment by MrGreedyPot** [comment]
Please add an option for automatic transparency like translucenttb
Window maximized = bars are opaque
Window mode or minimized = bars are transparent

**Comment by manmathew** [comment]
I've grown to like Mica quite a bit, but I have some hopes for MFE 2.
Could the Mica effect use what's behind the window as the canvas for the effect? Currently it only uses the wallpaper, but elements like the start menu that use Mica as a system default use whatever is behind it rather than just the wallpaper.
Also, would it be possible to make the gradient less blurred? I'm not sure if I'm expressing this correctly, but MFE Mica seems to have fewer changes in color over the gradient than the built-in Windows effect. The following 2 screenshots show the same area with the start menu mica and MFE mica so you know what I'm trying to get at.
[https://imgur.com/a/I8gdFEy](https://imgur.com/a/I8gdFEy)
[https://imgur.com/a/l79uIue](https://imgur.com/a/l79uIue)
p.s. idk why the start menu screenshot is marked as 18+ by imgur, it's perfectly safe so idk.
Also if the start menu uses acrylic and I've been misinformed, then could we get an acrylic effect that's strong like that? The current one is much more transparent if this is the case.

**Reply by dongle-the-gadget** [reply]
> Could the Mica effect use what's behind the window as the canvas for the effect

No, we still aren't supporting hooking for this current release.

> elements like the start menu

That's Acrylic, not Mica. Start menu also uses custom properties over their Acrylic effect over the default ones AFAIK.

**Comment by Void48** [comment]
@dongle-the-gadget Has there been any progress on MFE 2 or is development still paused? Just asking 🫡

**Reply by dongle-the-gadget** [reply]
There haven't been much going on unfortunately.

**Reply by Void48** [reply]
Take care in the meantime. Hopefully it'll be worth the wait ❤️
